include/cocaine/traits/attributes.hpp

```diff
@@ -60,32 +69,25 @@ struct type_traits<blackhole::attribute::value_t> {
 };
 
 template<>
-struct type_traits<blackhole::attribute::set_t> {
+struct type_traits<blackhole::attribute_t> {
+    template<class Stream>
     static inline
     void
-    unpack(const msgpack::object& source, blackhole::attribute::set_t& target) {
-        if(source.type != msgpack::type::MAP) {
-            throw msgpack::type_error();
-        }
-
-        msgpack::object_kv *ptr = source.via.map.ptr,
-                           *end = ptr + source.via.map.size;
-
-        for(; ptr < end; ++ptr) {
-            if(ptr->key.type != msgpack::type::RAW) {
-                throw msgpack::type_error();
-            }
-
-            const std::string& name = ptr->key.as<std::string>();
-
-            blackhole::attribute::value_t value;
-            type_traits<blackhole::attribute::value_t>::unpack(ptr->val, value);
+    pack(msgpack::packer<Stream>& packer, const blackhole::attribute_t& source) {
+        type_traits<
```

Comment:
I think this one and the next one are actually more readable when kept on one line.